### ISSUE TITLE
Request to add brightness and contrast adjustments

### ISSUE THREAD

**Issue by Jkapp76** [joncampbell123/dosbox-x#3461]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar feature request(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Is your feature request related to a problem? Please describe.

This request is not about a bad problem, just a feature to get closer to true hardware.

My true old CGA machine has much more vibrant colors than DOSBox has on my laptop screen.

The red on Castle Wolfenstein and Wheel of Fortune 2nd Edition are mind-blowing on the old CRT monitor and pretty dull on the laptop screen. The colors are thick and glowing.

I'd like some type of color adjustment for DOSBox, maybe just a brightness and contrast would do it. It seems like increasing the contrast would be close.

### What you want

I think this feature would allow us to get closer to emulating our true hardware.

**Comment by Keatah** [comment]
Try the ECE build. It isn't as feature rich as X, but the colors really wowed me on Tempest-2000 from the mid-1990's. A step in the right direction.

**Comment by BridgeHeadland** [comment]
I see that Castle Wolfenstein uses 4-color palette 0 (red, yellow, green, black background).

The way I see it, the color codes for each single color should be like this, normally:

#FF5555/Red = 255
#FF5555/Green = 85
#FF5555/Blue = 85
#FFFF55/Red = 255
#FFFF55/Green = 225
#FFFF55/Blue = 85
#55FF55/Red = 85
#55FF55/Green = 255
#55FF55/Blue = 85
#000000/Red = 0
#000000/Green = 0
#000000/Blue = 0

Probably, in your laptop's case, the color codes are like this:

#FF5555/Red = 252
#FF5555/Green = 84
#FF5555/Blue = 84
#FFFF55/Red = 252
#FFFF55/Green = 252
#FFFF55/Blue = 84
#55FF55/Red = 84
#55FF55/Green = 252
#55FF55/Blue = 84
#000000/Red = 0
#000000/Green = 0
#000000/Blue = 0

Which reduces the saturation by 8 times (from 240 to 232), and the brightness by 2 times (from 160 to 158). That's the case with my PC, and I think it's due to the graphics card in the device (I'm using Windows 10, by the way), from what I've heard. It's possible I'm wrong, I'm not a computer expert, you know.

**Comment by Jkapp76** [comment]
Yes. The colors look very pale on my modern PCs running DOSBox. They look dark and bold on my old CGA computer, very rich.

I sold both of my CGA computers recently or I'd upload screenshots.

**Comment by Torinde** [comment]
Related:
- https://github.com/dosbox-staging/dosbox-staging/pull/4618
